interfaces/Referral-App/src/app/referral/referral.page.ts

```diff
@@ -28,34 +31,65 @@ export class ReferralPage {
 
   public referralPageData: ReferralPageData = mockReferralPageData;
 
+  public readonly rootHref: string = '/';
+
   constructor(
     public offersService: OffersService,
     private route: ActivatedRoute,
     private router: Router,
     public modalController: ModalController,
     private loggingService: LoggingService,
     private referralPageDataService: ReferralPageDataService,
-  ) {
-    this.loadReferralData();
+  ) {}
+
+  ngOnInit() {
+    this.route.params.subscribe((params) => {
+      this.region = params['region'];
+      this.loadReferralData();
+    });
+  }
+
+  public getRegionHref() {
+    return this.rootHref + this.region;
   }
 
+  public isSupportedRegion() {
+    return (
+      this.region &&
+      this.supportedRegions.includes(
+        this.region.replace(/\-/g, ' ').toLowerCase(),
+      )
+    );
+  }
+
+  public toSnakeCase = (string) => {
+    return string
+      .replace(/\W+/g, ' ')
+      .split(/ |\B(?=[A-Z])/)
+      .map((word) => word.toLowerCase())
+      .join('-');
```

Comment:
Isn't this "kebab-case"? https://en.wikipedia.org/wiki/Letter_case#Special_case_styles